docs/dev/Lua API.rst

```diff
@@ -1580,6 +1580,22 @@ Units module
 
   Returns a table of the cutoffs used by the above stress level functions.
 
+Military module
+~~~~~~~~~~~~~~~~~~~
+
+* ``dfhack.military.makeSquad(assignment_id)``
+
+  Creates a new squad associated with the assignment. Fails if one already exists.
+  Note: This function does not name the squad, but they are otherwise complete.
```

Comment:
perhaps document how the caller is supposed to name the squad?

Reply:
also, what's an assignment and how to you get one?

Reply:
On names: It uses the general df language_name system, so I'm not 100% sure what the docs would look like here - just say eg "which can be modified via name"? Its worth noting that this uses df's language_name system, and that I'm not sure the reverse engineering has been done for how squad names are generated by the game

Reply:
On assignments: These are `entity_position_assignment`s, I can update the docs to reflect that but if it might be more helpful to actually document where/how to get one specifically I can do that too

Reply:
> I'm not sure the reverse engineering has been done for how squad names are generated by the game

ran into that code just the other day, what do you want to know about it?

Reply:
Oh! I'd love to know how they're generated. So each name has a `words` array, and I *think* for squads only [3] and [6] are actually used - how they're generated would be incredibly useful to know. I'd happily cram that into this PR so that we get game-equivalent randomly generated naming for squads here

I think also that the name for a squad is also generated with a `language` of 0, but I don't know that for a provable fact concretely

Overall I'm pretty unfamiliar with what the language word table is actually doing

Reply:
ok, here's the code that creates a new squad:
```
                local_88 = (squad *)operator_new(0x1e8);
                psVar24 = squad::_construct(local_88,'\0');
                local_80 = psVar24;
                historical_entity::name_randomizer
                          (plotinfo.main.fortress_entity,&psVar24->name,Squad);
```
`historical_entity::name_randomizer` picks a random name using the word tables for the specified historical entity (which in this case is the fortress entity). The `Squad` name type tells it that the name is being requested for a "squad". However, there is no specific rule for choosing names for squads so the selections are made from the entity's generic word list instead of one of the purpose-specific lists.

The process by which a random name is chosen is _quite_ complex, and while I've disassembled that code I can't claim I fully understand it and certainly can't give you a precise recap of the algorithm, at least not right now.

Reply:
I've expanded the docs here to make it more clear what the assignment is, and expanding on naming a bit. I could also add a bit on how to find the entity assignment you're looking for, but its not tremendously straightforward so it'd likely add a reasonable chunk to be complete. Its the kind of thing that might be much better in a general overview of the noble position/assignment system in general, but then again maybe something here is better than nothing